File: src/main/java/lan/dk/podcastserver/manager/worker/updater/NoOpUpdater.java
package lan.dk.podcastserver.manager.worker.updater;

import lan.dk.podcastserver.entity.Item;
import lan.dk.podcastserver.entity.Podcast;
import lan.dk.podcastserver.utils.facade.UpdateTuple;

import java.util.Set;
import java.util.function.Predicate;

/**
 * Created by kevin on 09/03/2016 for Podcast Server
 */
public class NoOpUpdater implements Updater {

    @Override
    public UpdateTuple<Podcast, Set<Item>, Predicate<Item>> update(Podcast podcast) { return null; }

    @Override
    public Set<Item> getItems(Podcast podcast) {
        return podcast.getItems();
    }

    @Override
    public String signatureOf(Podcast podcast) {
        return "";
    }

    @Override
    public Predicate<Item> notIn(Podcast podcast) {
        return item -> Boolean.FALSE;
    }

    @Override
    public AbstractUpdater.Type type() { return null; }

    @Override
    public Integer compatibility(String url) { return Integer.MAX_VALUE; }
}

File: backend/src/main/kotlin/com/github/davinkevin/podcastserver/manager/worker/noop/NoOpUpdater.kt
package com.github.davinkevin.podcastserver.manager.worker.noop

import com.github.davinkevin.podcastserver.manager.worker.*
import reactor.kotlin.core.publisher.toMono
import java.net.URI

/**
 * Created by kevin on 09/03/2016 for Podcast Server
 */
class NoOpUpdater : Updater {

    override fun update(podcast: PodcastToUpdate) = NO_MODIFICATION.toMono()

    override fun blockingFindItems(podcast: PodcastToUpdate) = setOf<ItemFromUpdate>()

    override fun blockingSignatureOf(url: URI) = ""

    override fun type() = Type("NoOpUpdater", "NoOpUpdater")

    override fun compatibility(url: String?) = Integer.MAX_VALUE
}
